cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/workflow/DefaultWorkflowConfigurer.java

```diff
@@ -112,8 +139,7 @@ private WorkflowNode createNodeWithId(WorkflowNode node) {
     WorkflowNode nodeWithId = null;
     switch (node.getType()) {
       case ACTION:
-        nodeWithId = createActionNodeWithId(node);
-        break;
+        return node;
```

Comment:
Previously when nodes were being configured, we used to assign the `null` to the node id. Node ids were generated using `createNodesWithId(nodes)` before creating the specifications for the Workflow. This helped having single counter `nodeIdentifier` in the `DefaultWorkflowConfigurer`.

Now since the nodeId is the name of the program or unique name, while configuring the node itself we assign the nodeId to it. So the `createNodeWithId` method returns the node that was passed to it as a parameter if it is of type 'ACTION', since the node id is already assigned to it. 

However for fork nodes (added directly or created inside the condition nodes), we still need to assign the node ids, which is done in this method using `nodeIdentifier`.